### DISCUSSION TITLE
Sorting of data picker items by name

### DISCUSSION THREAD

**Discussion by erlando** [leekelleher/umbraco-contentment#365] [Question]
Hi.

Love the package.

Is it possible to get the data picker to sort the items by DataListItem.Name? I've tried to sort the items returned by IDataPickerSource.GetItemsAsync but it doesn't seem to have any effect.

<img width="632" alt="image" src="https://github.com/leekelleher/umbraco-contentment/assets/381754/50215bc7-71ce-4adc-ae59-38fd167972d9">

<img width="678" alt="image" src="https://github.com/leekelleher/umbraco-contentment/assets/381754/7c824106-89d1-4a2e-aac7-210ffa75f827">

Also, is it possible to hide the id of the items? The users for the custom section that this is being displayed in are bound to get confused by having an Udi displayed like that.

Best regards, Søren

**Comment by leekelleher** [comment]
Hi @erlando,

The `IDataPickerSource.GetItemsAsync()` method is used to retrieve a metadata lookup (e.g. name, description, icon), of the saved values (for the initial page load of the content node), it doesn't make any changes to the saved value itself.

If you need to manipulate the underlying property value, you might want to hook into a notification event, (something like [content saving](https://docs.umbraco.com/umbraco-cms/reference/notifications/contentservice-notifications), or the [editor model sending](https://docs.umbraco.com/umbraco-cms/reference/notifications/editormodel-notifications)).

In terms of your code snippet, is this your own custom data-source code? (only that I don't recognise it).  For hiding the member's UDI, that should only be when the `DataListItem.Description` has been set, (I don't think there's a fallback on the `.Value` 🤔).

**Comment by erlando** [comment]
Hi,

Yes, it's my own custom code. It fetches data from custom tables in the database and looks up member-udi's via the API. There must be a fallback of some kind because if I set the Description to `" "` (not `""`) in `GetItemsAsync` the Value is no longer displayed. That works for me. Thank you :-)

I can't hook into any notifications because it's actually not Umbraco that is rendering the form. I'm building the form myself from a Document Type using `<umb-property-editor>` and use that form to edit data in custom tables. Even if I sort the data client-side before supplying it to the data picker editor it's still not sorted when displayed. It's a bit weird.
